### PR TITLE
Table column sorting toggleSort function takes in magic number #4709

### LINKED ISSUE #4709: Table column sorting toggleSort function takes in magic number
Currently, the toggleSort function for sorting table rows in various places takes in a magic number corresponding to the column position. Should find a way to improve the function such that the magic number is not required.

### PR DESCRIPTION
**Pull request by unyoungwax**
Fixes #4709

### CODE REVIEW

**Comment by wkurniawan07** [comment]
@unyoungwax you can add the `s.ToReview` label yourself next time. Now I'm giving you the freedom to arrow someone to review :p (don't arrow me please...)

**Comment by unyoungwax** [comment]
I think this pr falls under the code quality category, so I am afraid that I have to arrow you @wkurniawan07.

File: src/main/webapp/WEB-INF/tags/admin/sessions/institutionPanel.tag [hunk 1]
```diff
@@ -33,7 +33,7 @@
                     <th onclick="toggleSort(this,5,sortDate)"
```
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/admin/sessions/institutionPanel.tag]: See if you can remove the `4` and `5` here also, and make `sortDate` the 2nd parameter?

File: src/main/webapp/WEB-INF/tags/instructor/home/courseTable.tag [hunk 2]
```diff
@@ -6,7 +6,7 @@
 <table class="table-responsive table table-striped table-bordered">
     <thead>
         <tr>
-            <th onclick="toggleSort(this,1);" class="button_sortname button-sort-none">
+            <th onclick="toggleSort(this);" class="button_sortname button-sort-none">
                 Session Name<span class="icon-sort unsorted"></span></th>
             <th onclick="toggleSort(this,2,instructorHomeDateComparator);" class="button_sortstartdate button-sort-none">Start Date<span class="icon-sort unsorted"></span></th>
             <th onclick="toggleSort(this,3,instructorHomeDateComparator);" class="button_sortenddate button-sort-none">End Date<span class="icon-sort unsorted"></span></th>
```
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/instructor/home/courseTable.tag]: Likewise here.

File: src/main/resources/feedbackQuestionContribResultStatsTemplate.html [hunk 3]
```diff
@@ -14,9 +14,9 @@
...
                 <thead>
                 <tr>
-                    <td class="button-sort-ascending" id="button_sortteamname" onclick="toggleSort(this,1);">Team 
+                    <td class="button-sort-ascending" id="button_sortteamname" onclick="toggleSort(this);">Team 
                         <span class="icon-sort unsorted"></span></td>
-                    <td class="button-sort-none" id="button_sortname" onclick="toggleSort(this,2)">Student
+                    <td class="button-sort-none" id="button_sortname" onclick="toggleSort(this)">Student
                         <span class="icon-sort unsorted"></span></td>
                     <td class="button-sort-none" id="button_sortclaimed" onclick="toggleSort(this,3,sortByPoint)" data-toggle="tooltip" data-placement="top" data-container="body" title='${Const.Tooltips.CLAIMED}'>
                         <abbr title="Claimed Contribution">CC</abbr>
```
Comment (wkurniawan07) [inline src/main/resources/feedbackQuestionContribResultStatsTemplate.html]: Likewise here.

File: src/main/resources/feedbackQuestionNumScaleResultStatsTemplate.html [hunk 4]
```diff
@@ -11,9 +11,9 @@
             <table class="table table-bordered table-responsive margin-0">
                 <thead>
                 <tr>
-                    <td class="button-sort-ascending" id="button_sortreceiverteam" onclick="toggleSort(this,1);">Team 
+                    <td class="button-sort-ascending" id="button_sortreceiverteam" onclick="toggleSort(this);">Team 
                         <span class="icon-sort unsorted"></span></td>
-                    <td class="button-sort-ascending" id="button_sortreceivername" onclick="toggleSort(this,2);">Recipient 
+                    <td class="button-sort-ascending" id="button_sortreceivername" onclick="toggleSort(this);">Recipient 
                         <span class="icon-sort unsorted"></span></td>
                     <td class="button-sort-none" id="button_sortavg" onclick="toggleSort(this,3,sortByPoint)" data-toggle="tooltip" data-placement="top" data-container="body" title="Average of the visible responses">
```
Comment (wkurniawan07) [inline src/main/resources/feedbackQuestionNumScaleResultStatsTemplate.html]: Here, too.

File: src/main/resources/feedbackQuestionNumScaleResultStatsTemplateWithSelfResponse.html [hunk 5]
```diff
@@ -11,9 +11,9 @@
             <table class="table table-bordered table-responsive margin-0">
                 <thead>
                 <tr>
-                    <td class="button-sort-ascending" id="button_sortreceiverteam" onclick="toggleSort(this,1);">Team 
+                    <td class="button-sort-ascending" id="button_sortreceiverteam" onclick="toggleSort(this);">Team 
                         <span class="icon-sort unsorted"></span></td>
-                    <td class="button-sort-ascending" id="button_sortreceivername" onclick="toggleSort(this,2);">Recipient 
+                    <td class="button-sort-ascending" id="button_sortreceivername" onclick="toggleSort(this);">Recipient 
                         <span class="icon-sort unsorted"></span></td>
                     <td class="button-sort-none" id="button_sortavg" onclick="toggleSort(this,3,sortByPoint)" data-toggle="tooltip" data-placement="top" data-container="body" title="Average of the visible responses">
```
Comment (wkurniawan07) [inline src/main/resources/feedbackQuestionNumScaleResultStatsTemplateWithSelfResponse.html]: Here, too. I hope this is the last?

File: src/main/webapp/js/common.js [hunk 6]
```diff
@@ -169,8 +169,15 @@ function isTouchDevice() {
  *     The column index (0-based) e.g. use 2 if <th> has 2 rows so that the headers are not sorted.
  */
 function toggleSort(divElement, colIdx, comparator, row) {
```
Comment (wkurniawan07) [inline src/main/webapp/js/common.js]: The 4th parameter `row` seems unused. If that's really the case, you can remove it altogether.
While you're at it, also update the header comment of this method to sync with the new parameters.

**Comment by wkurniawan07** [comment]
On top of those, try a project-wide search to check that you don't miss any of the `toggleSort` functions in some files.

**Comment by unyoungwax** [comment]
I have removed some parameters (`colIdx` and `row`) from the toggleSort function and updated the comment.

**Comment by wkurniawan07** [comment]
Looks good :+1:

**Pull request merged**